Crypto/PublicKey+Apple.mm

```diff
@@ -682,15 +681,47 @@ virtual int _sign(int/*mbedtls_md_type_t*/ mbedDigestAlgorithm,
             
             CFIndex count = SecTrustGetCertificateCount(trustRef);
             Assert(count > 0);
+            if (count == 1) {
+                NSDictionary* params = @{
+                    (id)kSecClass:              (id)kSecClassCertificate,
+                    (id)kSecValueRef:           (__bridge id)certRef
+                };
+                checkOSStatus(SecItemDelete((CFDictionaryRef)params),
+                              "SecItemDelete",
+                              "Couldn't delete a certificate from the Keychain");
+                return;
+            }
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 120000 || __IPHONE_OS_VERSION_MAX_REQUIRED >= 150000
             if (@available(macOS 12.0, iOS 15.0, *)) {
                 CFArrayRef certs = SecTrustCopyCertificateChain(trustRef);
                 for (CFIndex i = count - 1; i >= 0; i--) {
-                    SecCertificateRef ref = (SecCertificateRef)CFArrayGetValueAtIndex(certs, i);
-                    if (getChildCertCount(ref) < 2) {
-                        NSDictionary* params = @{
+                    SecCertificateRef copiedRef = (SecCertificateRef)CFArrayGetValueAtIndex(certs, i);
+                    if (getChildCertCount(copiedRef) < 2) {
+                        // Get public key hash from kSecAttrApplicationLabel attribute:
+                        SecKeyRef publicKeyRef = SecCertificateCopyKey(copiedRef);
+                        NSDictionary* pubKeyAttrs = CFBridgingRelease(SecKeyCopyAttributes(publicKeyRef));
+                        NSData* publicKeyHash = [pubKeyAttrs objectForKey: (id)kSecAttrApplicationLabel];
+                        CFRelease(publicKeyRef);
+                        
+                        // For certs, primary key: issuer + serial-num + cert-type
+                        NSDictionary* attrs = CFBridgingRelease(findInKeychain(@{
                             (id)kSecClass:              (id)kSecClassCertificate,
-                            (id)kSecValueRef:           (__bridge id)ref
+                            (id)kSecValueRef:           (__bridge id)copiedRef,
+                            (id)kSecReturnAttributes:   @YES
+                        }));
+                        NSString* issuer = [attrs objectForKey: (id)kSecAttrIssuer];
+                        Assert(issuer);
+                        NSString* serialNum = [attrs objectForKey: (id)kSecAttrSerialNumber];
+                        Assert(serialNum);
+                        NSNumber* certType = [attrs objectForKey: (id)kSecAttrCertificateType];
+                        Assert(certType != nil);
+                        
+                        NSDictionary* params = @{
+                            (id)kSecClass:                  (__bridge id)kSecClassCertificate,
+                            (id)kSecAttrPublicKeyHash:      publicKeyHash,
```

Comment:
Can you remove public key hash from the query as the primary key is primary key: issuer + serial-num + cert-type?